lib/ddtrace/contrib/mysql2/patcher.rb

```diff
@@ -20,7 +21,15 @@ def patch
         end
 
         def patch_mysql2_client
-          ::Mysql2::Client.include(Instrumentation)
+          # ::Mysql2::Client.include(Instrumentation::Client)
+
+          Datadog::Instrumentation::Hook['Mysql2::Client#query'].add do
```

Comment:
Would it make sense to use the class name directly here? E.g. `Hook[Mysql2::Client, '#query']`?

Reply:
This would cause an exception if the constant does not exist (e.g the gem is not in the bundle). `Hook` handles that internally and `add` (+`install` IIRC) becomes NOOP.

Reply:
Yes...ish. The current way that dd-trace-rb does it is that there's a separate check for that in `integration.rb`: https://github.com/DataDog/dd-trace-rb/blob/87c30b0316908d67f669a7066404d2d59ab24afc/lib/ddtrace/contrib/mysql2/integration.rb#L18
which is why the old code referenced the constant directly. 

Using it as a string is good if we could get rid of that separate file, otherwise, we already get the validation for free.